### ISSUE TITLE
As a peer pods tester I want the ability to run peer pod enablement on an existing cluster

### ISSUE THREAD

**Issue by Jonah-F** [confidential-containers/cloud-api-adaptor#22]
## Description
We want to split the Ansible playbook in `terraform/cluster/ansible` into two new Ansible playbooks so we have the benefit running one for the initial cluster setup, and another for Peer Pod enablement. 

## Use case
_As a peer pods tester I want the ability to run peer pod enablement on an existing cluster. This will be the setup for an operator that runs on a Kube cluster._

## Task List
- [x] In the `terraform/cluster/ansible` playbook, split the initial cluster setup code from the Peer Pod enablement code into two playbooks.
- [x] Determine how the two new playbooks are going to run, as to not disrupt the current demo instructions.
- [x] Change the Terraform template to run the two playbooks in the correct order.
- [x] Determine how the README in `terraform/`should be updated and update the README.
- [x] Raise a PR with the changes. The PR must be approved and merged before this story is closed.

## Performance Considerations
Increasing the number of manual steps goes against the idea of automation, when we are splitting a playbook into two to have control over which functionality we want, ensure that the steps to run the demo aren't made more complicated.

## Security Considerations 
API keys must not be shared or pushed to this repository.

### Questions
1. Are the two Ansible playbooks generated going to run back to back?

2. How would this issue be documented in the `/terraform/README`. We need to decide if the instructions to run the README to setup the demo will evolve into something new as the functionality of the Terraform scripts evolve.
 
3. How do we capture the benefits of one without increasing the number of steps for the demo? 

## Acceptance Criteria
**Given** I want to run Peer Pod enablement on an existing cluster
**When** I split the existing `/cluster/ansible/`  playbook up into separate functionality
**Then** the existing Terraform template can run the new Peer Pod enablement Ansible playbook against an existing cluster without running the initial cluster setup steps

**Scenario: Regression testing**
**Given** I want to create a new Kubernetes cluster and enable it with the peer-pod capability
**When** I follow the `Create a Kubernetes cluster` and peer pod enablement documentation 
**Then** I have a new, working Kubernetes cluster and it has the kata-shim, cloud-api-adaptor and all the required dependencies installed on it

**Given** I want to be able to create an x86 and s390x cluster
**When** the existing Ansible playbook is split into two and I run them
**Then** I maintain the option to choose an x86 or s390x cluster with peer-pod enablement for creation

**Comment by Jonah-F** [comment]
Hi @yoheiueda, the `terraform/cluster/ansible` playbook lines 230-250 include some code I’m unsure if It belongs as a Kata or Kubernetes dependency. The code can be seen here:
```
 - name: Copy systemd unit file for fix-resolvconf
      copy:
        dest: /etc/systemd/system/fix-resolvconf.service
        content: |
          [Unit]
          Description=Make /etc/resolv.conf a regular file
          After=systemd-networkd.service
          [Service]
          ExecStart=/bin/bash -c 'rm -f /etc/resolv.conf && cp /run/systemd/resolve/resolv.conf /etc/resolv.conf'
          ExecStop= /bin/bash -c 'rm -f /etc/resolv.conf && ln -s ../run/systemd/resolve/stub-resolv.conf /etc/resolv.conf'
          Type=oneshot
          RemainAfterExit=yes
          [Install]
          WantedBy=multi-user.target
    - name: Start fix-resolvconf service
      systemd:
        name: fix-resolvconf.service
        state: started
```
To fulfil this story's requirements I need to split the existing playbook up into the Peer-Pod Enablement and the Initial Cluster Setup pieces, knowing what which dependencies belong in which playbook is critical to the progress of this story, your input would be very helpful. Thanks!

**Comment by yoheiueda** [comment]
@Jonah-F This code belongs to Kata.

This is a workaround fix for an issue related to name resolutions in a pod VM. Without this fix, kata agent fails to resolve container repository names such as docker.io, and can't pull caontainer images.

**Comment by Jonah-F** [comment]
I've split the existing playbook.yml into two new ones. One called kube-playbook.yml, the other kata-playbook.yml.
Steve and I flagged sections of the existing playbook into Kata and Kube related pieces. I then took those pieces and split them into their new respective playbook. 

I've run the two new playbooks separately, both with a successful exit.
I've run them as consecutively, kube first then kata, both with a successful exit.
I've run them as part of the existing Terraform template, and after fixing a lack of including the `runc`, it exited with a successful exit.

From the worker node post `terraform apply` using both new playbooks in order:
```
$ kubectl get nodes
NAME                    STATUS   ROLES           AGE     VERSION
jf-kube-test-2-cp       Ready    control-plane   3m24s   v1.24.0
jf-kube-test-2-worker   Ready    <none>          2m49s   v1.24.0
```
